purchases/src/main/kotlin/com/revenuecat/purchases/google/BillingWrapper.kt

```diff
@@ -461,33 +459,33 @@ internal class BillingWrapper(
         onCompletion: (StoreTransaction) -> Unit,
         onError: (PurchasesError) -> Unit,
     ) {
-        withConnectedClient {
-            log(LogIntent.DEBUG, RestoreStrings.QUERYING_PURCHASE_WITH_TYPE.format(productId, productType.name))
-            productType.toGoogleProductType()?.let { googleProductType ->
-                queryPurchaseHistoryAsyncEnsuringOneResponse(googleProductType) { result, purchasesList ->
-                    if (result.isSuccessful()) {
-                        val purchaseHistoryRecordWrapper =
-                            purchasesList?.firstOrNull { it.products.contains(productId) }
-                                ?.toStoreTransaction(productType)
-
-                        if (purchaseHistoryRecordWrapper != null) {
-                            onCompletion(purchaseHistoryRecordWrapper)
-                        } else {
-                            val message = PurchaseStrings.NO_EXISTING_PURCHASE.format(productId)
-                            val error = PurchasesError(PurchasesErrorCode.PurchaseInvalidError, message)
-                            onError(error)
-                        }
+        log(LogIntent.DEBUG, RestoreStrings.QUERYING_PURCHASE_WITH_TYPE.format(productId, productType.name))
+        productType.toGoogleProductType()?.let { googleProductType ->
+            QueryPurchaseHistoryUseCase(
```

Comment:
I wonder if we should move tests using `findPurchaseInPurchaseHistory` also to this use case's test class...

Reply:
Let's do it